server/src/controllers/appointments.controller.js

```diff
@@ -1,29 +1,50 @@
 import Appointment from "../models/appointments.model.js";
 import appointmentSchema from "../validators/appointments.validator.js";
+import emailHelper, {
+  apptRequestConfirmationHtml,
+  apptRequestConfirmationText,
+} from "../utils/emailHelper.js";
 
 export async function newAppointment(req, res, next) {
   try {
+    // handle validation errors
     const { error } = appointmentSchema.validate(req.body);
 
-    // handle validation errors
     if (error) {
       res.status(422);
       return next({ message: error.details[0].message });
     }
 
-    // send value to db
-    const { earlyTimeHour, lateTimeHour, ...rest } = req.body;
-    const newAppt = new Appointment({
+    const { earlyTimeHour, lateTimeHour, email, ...rest } = req.body;
+
+    // add request data to the database
+    const appt = new Appointment({
       ...rest,
+      email,
       userId: "Insert Id Here",
       timeRange: {
         earlyTimeHour,
         lateTimeHour,
       },
     });
-    await newAppt.save();
 
-    // send success response
+    const newAppt = await appt.save();
+
+    // send mock appt request email, add url to the database
+    const emailPreviewUrl = await emailHelper({
+      email,
+      subject: "Appointment Request Received",
+      text: apptRequestConfirmationText(req.body),
+      html: apptRequestConfirmationHtml(req.body),
+    });
```

Comment:
The signature here feels a little odd. emailHelper.js exposes these 2 methods to get the text and html, but then we only use those as for parameters into the other function it exposes.

```suggestion
    const emailPreviewUrl = await emailHelper({
      email,
      subject: "Appointment Request Received",
      formData: req.body
    });
```
If you had something like this then it seems better encapsulated and more cohesive? `emailHelper` is also not very descriptive, would `sendEmail` be an appropriate name?

Reply:
Yeah I was struggling with the signature, but can't come up with a better way while still keeping `sendEmail` generic, or moving the text and email generation into the controller function itself